src/Raven.Server/Documents/PeriodicBackup/RavenFtpClient.cs

```diff
@@ -65,135 +67,121 @@ public void UploadFile(string folderName, string fileName, Stream stream)
+                            client.CreateDirectory($"/{directory}");
+                    }
+                    catch (WebException e)
+                    {
+                        throw ;
                     }
-
-                    throw;
                 }
             }
 
             return url;
         }
 
-        private void ExtractUrlAndDirectories(out string url, out List<string> dirs)
+        private void ExtractUrlAndDirectories(out List<string> dirs)
         {
             var uri = new Uri(_url);
 
             dirs = uri.AbsolutePath.TrimStart('/').TrimEnd('/').Split("/").ToList();
-            var port = _port ?? (uri.Port > 0 ? uri.Port : DefaultFtpPort);
-            if (port < 1 || port > 65535)
-                throw new ArgumentException("Port number range: 1-65535");
-
-            url = $"{uri.Scheme}://{uri.Host}:{port}";
         }
 
-        private FtpWebRequest CreateFtpWebRequest(string url, string method, bool keepAlive)
+        internal FtpClient CreateFtpClient(string url, bool keepAlive)
         {
-            var request = (FtpWebRequest)WebRequest.Create(new Uri(url));
-            request.Method = method;
-
-            request.Credentials = new NetworkCredential(_userName, _password);
-            request.EnableSsl = _useSsl;
+            var client = new FtpClient(url);
+            client.Config.ConnectTimeout = -1;
+            client.Credentials = new NetworkCredential(_userName, _password);
             if (_useSsl)
             {
                 var byteArray = Convert.FromBase64String(_certificateAsBase64);
 
                 try
                 {
-                    var x509Certificate = new X509Certificate(byteArray);
-                    request.ClientCertificates = new X509CertificateCollection(new[] { x509Certificate });
+                    var x509Certificate = new X509Certificate2(byteArray);
+                    client.Config.EncryptionMode = FtpEncryptionMode.Explicit;
+                    client.Config.SslProtocols = SslProtocols.Tls12 | SslProtocols.Tls13;
+                    client.Config.ClientCertificates.Add(x509Certificate);
+                    client.ValidateCertificate += (control, e) =>
+                    {
+                        e.Accept = e.PolicyErrors == SslPolicyErrors.None;
+                    };
                 }
                 catch (Exception e)
                 {
                     throw new ArgumentException($"This is not a valid certificate, file name: {_certificateFileName}", e);
                 }
             }
-                
-            request.UsePassive = true;
-            request.UseBinary = true;
-            request.KeepAlive = keepAlive;
 
-            return request;
+            client.Config.SocketKeepAlive = keepAlive;
+            if (_port != null)
+                client.Port = (int)_port;
+            else
+                client.Port = DefaultFtpPort;
+            client.Connect();
```

Comment:
IRIC there is a method AutoConnect or sth like that, maybe it is better to use it? What it does?

Reply:
It's automatically searching for FTP settings and if it finds one it automatically connecting otherwise returning a null, I will convert to AutoConnect method

Reply:
AutoConnect caused problems with the tests were Connect didn't